mmeds/database/database.py

```diff
@@ -967,9 +968,41 @@ def check_study_name(self, study_name):
         """ Verifies the provided study name is valid and not already in use. """
         if not study_name.replace('_', '').isalnum():
             raise StudyNameError("Only alpha numeric characters and '_' are allowed in the study name")
-        if MMEDSDoc.objects(study_name=study_name):
+        if MMEDSDoc.objects(study_name=study_name, doc_type='study'):
             raise StudyNameError(f"Study name {study_name} already in use")
 
+    def check_sequencing_run_name(self, run_name):
+        """ Verifies the provided sequencing run name is valid and not already in use. """
+        if not run_name.replace('_', '').isalnum():
+            raise StudyNameError("Only alpha-numeric characters and '_' are allowed in the sequencing run name")
+        if MMEDSDoc.objects(study_name=run_name, doc_type='sequencing_run'):
+            raise StudyNameError(f"Sequencing Run name {run_name} already in use")
+
+    def get_sequencing_run_locations(self, metadata, user, column=("RawDataProtocol", "RawDataProtocolID")):
+        """ Returns the list of sequencing runs as a dict of dir paths """
+        df = pd.read_csv(metadata, sep='\t', header=[0, 1], skiprows=[2, 3, 4])
+
+        # Store run names from metadata
+        runs = []
+        for run in df[column]:
```

Comment:
list and dictionary comprehension might look nicer here, for filling up lists and dictionaries. Not the most important thing though